src/pal/src/numa/numa.cpp

```diff
@@ -0,0 +1,683 @@
+                CPU_SET(cpu, &cpuSet);
+            }
+        }
+    }
+
+    int st = pthread_setaffinity_np(thread, sizeof(cpu_set_t), &cpuSet);
+
+    if (st == -1)
+    {
+        switch (errno)
+        {
+            case EINVAL:
+                // There is no processor in the mask that is allowed to execute the process
+                SetLastError(ERROR_INVALID_PARAMETER);
+                break;
+            case EPERM:
+                SetLastError(ERROR_ACCESS_DENIED);
+                break;
+            default:
+                SetLastError(ERROR_GEN_FAILURE);
+                break;
+        }
+    }
+
+    BOOL success = (st == 0);
+#else // HAVE_PTHREAD_GETAFFINITY_NP
+    // There is no API to manage thread affinity, so let's ignore the request
+    BOOL success = TRUE;
+#endif // HAVE_PTHREAD_GETAFFINITY_NP
+
+    LOGEXIT("SetThreadGroupAffinity returns BOOL %d\n", success);
+    PERF_EXIT(SetThreadGroupAffinity);
+
+    return success;
+}
+
+/*++
+Function:
+  GetCurrentProcessorNumberEx
+
+See MSDN doc.
+--*/
+VOID
+PALAPI
+GetCurrentProcessorNumberEx(
+  OUT PPROCESSOR_NUMBER ProcNumber
+)
+{
+    PERF_ENTRY(GetCurrentProcessorNumberEx);
+    ENTRY("GetCurrentProcessorNumberEx(ProcNumber=%p\n", ProcNumber);
+
+    DWORD cpu = GetCurrentProcessorNumber();
+    _ASSERTE(cpu < g_cpuCount);
+    ProcNumber->Group = g_cpuToAffinity[cpu].Group;
+    ProcNumber->Number = g_cpuToAffinity[cpu].Number;
+
+    LOGEXIT("GetCurrentProcessorNumberEx\n");
+    PERF_EXIT(GetCurrentProcessorNumberEx);
+}
+
+/*++
+Function:
+  GetProcessAffinityMask
+
+See MSDN doc.
+--*/
+BOOL
+PALAPI
+GetProcessAffinityMask(
+  IN HANDLE hProcess,
+  OUT PDWORD_PTR lpProcessAffinityMask,
+  OUT PDWORD_PTR lpSystemAffinityMask
+)
+{
+    PERF_ENTRY(GetProcessAffinityMask);
+    ENTRY("GetProcessAffinityMask(hProcess=%p, lpProcessAffinityMask=%p, lpSystemAffinityMask=%p\n", hProcess, lpProcessAffinityMask, lpSystemAffinityMask);
+
+    BOOL success = FALSE;
+
+    if (hProcess == GetCurrentProcess())
+    {
+        DWORD_PTR systemMask = GetFullAffinityMask(g_cpuCount);
+
+#if HAVE_SCHED_GETAFFINITY
+        int pid = getpid();
+        cpu_set_t cpuSet;
+        int st = sched_getaffinity(pid, sizeof(cpu_set_t), &cpuSet);
+        if (st == 0)
+        {
+            WORD group = NO_GROUP;
+            DWORD_PTR processMask = 0;
+
+            for (int i = 0; i < g_cpuCount; i++)
+            {
+                if (CPU_ISSET(i, &cpuSet))
+                {
+                    WORD g = g_cpuToAffinity[i].Group;
+                    if (group == NO_GROUP || g == group)
+                    {
+                        group = g;
+                        processMask |= ((DWORD_PTR)1) << g_cpuToAffinity[i].Number;
+                    }
+                    else
+                    {
+                        // The process has affinity in more than one group, in such case
+                        // the function needs to return zero in both masks.
```

Comment:
The comment here says we need to return 0 in both masks, but unless I'm missing something we're not setting `systemMask` to 0, right?